OpenRA.Mods.Common/Widgets/ObserverProductionIconsWidget.cs

```diff
@@ -134,10 +134,11 @@ public override void Draw()
 					continue;
 
 				var rsi = actor.TraitInfo<RenderSpritesInfo>();
-				var icon = new Animation(world, rsi.GetImage(actor, faction));
+				var image = rsi.GetImage(actor, faction);
+				var icon = new Animation(world, image);
 				var bi = actor.TraitInfo<BuildableInfo>();
+				icon.Play(bi.GetIconSequence(world, GetPlayer().Faction.InternalName, image));
```

Comment:
```suggestion
				icon.Play(bi.GetIconSequence(world, player.Faction.InternalName, image));
```

the variable is already cached